networkd/network_monitor_thread.ml

```diff
@@ -299,13 +299,28 @@ let ip_watcher () =
     end;
     loop ()
   in
-  while true do
-    try
-      info "(Re)started IP watcher thread";
-      loop ()
-    with e ->
-      warn "Error in IP watcher: %s\n%s" (Printexc.to_string e) (Printexc.get_backtrace ())
-  done
+  try
+    while true do
+      try
+        info "(Re)started IP watcher thread";
+        loop ()
+      with e -> begin
+        warn "Error in IP watcher: %s\n%s" (Printexc.to_string e) (Printexc.get_backtrace ());
+        Thread.delay 5.0;
```

Comment:
I think this is not the best place because now we will introduce a gap in the log file. I think it is better to do this at the end of `with e -> begin ...; Thread.delay 5.0; end`.

Reply:
So move the delay right before next recursive call.